relay-server/src/actors/outcome.rs

```diff
@@ -296,9 +316,60 @@ impl Message for TrackRawOutcome {
     type Result = Result<(), OutcomeError>;
 }
 
+/// Common implementation for both processing and non-processing
+impl OutcomeProducer {
+    fn send_batch(&mut self, context: &mut Context<Self>) {
+        log::trace!("sending outcome batch");
+
+        //the future should be either canceled (if we are called with a full batch)
+        // or already called (if we are called by a timeout)
+        self.send_outcomes_future = None;
+
+        if self.unsent_outcomes.is_empty() {
+            log::warn!("unexpected send_batch scheduled with no outcomes to send.");
+            return;
+        }
+
+        let request = SendOutcomes {
+            outcomes: self.unsent_outcomes.drain(..).collect(),
```

Comment:
I would just swap the vector with a new one, should be cheaper than moving individual TrackRawOutcome into a new vec

Reply:
Agreed, `mem::take` is the way to go here.

Reply:
cool